website/mailing_list/utils.py

```diff
@@ -0,0 +1,422 @@
+    """
+    from website.models import Node  # avoid circular import
+    node = Node.load(node_id)
+
+    resp = requests.put(
+        '{}/{}'.format(MAILGUN_BASE_LISTS_URL, address(node_id)),
+        auth=('api', settings.MAILGUN_API_KEY),
+        data={
+            'name': list_title(node)
+        }
+    )
+    if resp.status_code != 200:
+        raise HTTPError(resp.status_code, data={'message_long': resp.json() or ''})
+
+@require_mailgun
+def update_single_user_in_list(node_id, user_id, email=None, enabled=True, old_email=None):
+    """ Adds/updates single member of a mailing list on Mailgun
+    Called to add, subscribe, or unsubscribe a user.
+
+    :param str node: ID of node in question
+    :param str user: ID of user to update
+    :param str email: email address to add. If None, `user.username` assumed.
+    :param bool enabled: Enable or disable user?
+    :param str old_email: Previous email of this user in list, included when user changes primary email.
+    """
+    from website.models import Node, User  # avoid circular import
+    node = Node.load(node_id)
+    user = User.load(user_id)
+    email = email or user.username
+
+    if old_email:
+        resp = requests.put(
+            '{}/{}/members/{}'.format(MAILGUN_BASE_LISTS_URL, address(node_id), old_email),
+            auth=('api', settings.MAILGUN_API_KEY),
+            data={
+                'subscribed': 'no',
+                'vars': json.dumps({'_id': user_id, 'primary': False})
+            }
+        )
+        if resp.status_code not in [200, 404]:
+            raise HTTPError(resp.status_code, data={'message_long': resp.json() or ''})
+
+    resp = requests.post(
+        '{}/{}/members'.format(MAILGUN_BASE_LISTS_URL, address(node._id)),
+        auth=('api', settings.MAILGUN_API_KEY),
+        data={
+            'address': email,
+            'subscribed': enabled and email == user.username and user not in get_unsubscribes(node),
+            'vars': json.dumps({'_id': user._id, 'primary': email == user.username or bool(old_email)}),
+            'upsert': True
+        }
+    )
+    if resp.status_code != 200:
+        raise HTTPError(resp.status_code, data={'message_long': resp.json() or ''})
+
+@require_mailgun
+def remove_user_from_list(node_id, user_id):
+    """ Removes single member of a mailing list on Mailgun
+    Called when a contributor is removed from a Node.
+
+    :param str node_id: ID of node in question
+    :param str user_id: ID of user to remove
+    """
+    from website.models import User  # avoid circular import
+    user = User.load(user_id)
+
+    for email in user.emails:
+        resp = requests.delete(
+            '{}/{}/members/{}'.format(MAILGUN_BASE_LISTS_URL, address(node_id), email),
+            auth=('api', settings.MAILGUN_API_KEY)
+        )
+        if resp.status_code not in [200, 404]:
+            raise HTTPError(resp.status_code, data={'message_long': resp.json() or ''})
```

Comment:
What kind of data does mailgun put in its error response payloads?

I *think* this function only gets called in celery... but by raising an HttpError, you're sending the entire error response payload out to whoever made the request. If this function ever connects to something in the OSF website front-end, we would want to avoid sending potentially sensitive data to the website user.  (who shouldn't even have to know we're using mailgun at all, much less see a detailed mailgun error)